### PR TITLE
add migrator for moving relevant_protocols to protocol_link PR21

### PR DESCRIPTION
**Pull request by brynnz22**
This PR provides the migration code that addresses PR: https://github.com/microbiomedata/berkeley-schema-fy24/pull/21

It moves any links that currently exist in the `relevant_protocols` slot over to the `protocol_link` slot and removes `relevant_protocols` from the study. 

This migration is not dependent on any others.

### CODE REVIEW

File: nmdc_schema/migrators/migrator_from_X_to_PR21.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        super().__init__(*args, **kwargs)
+
+        # Populate the "collection-to-transformers" map for this specific migration.
+        self.agenda = dict(
+            study_set=[self.move_relevant_protocols_to_protocol_link],
+        )
+
+    def move_relevant_protocols_to_protocol_link(self, study: dict):
+        r"""
+        Transforms `websites` values that are dois into curies and moves them into the `associated_dois slot
```
Comment (eecavanna) [inline nmdc_schema/migrators/migrator_from_X_to_PR21.py:19]: I think the `associated_dois` is missing the closing backtick.

**Review by eecavanna — APPROVED** [review]
Looks good to me. Thanks for summarizing the changes in the PR description.

Note: This migrator has the "original" structure because the recent upstream changes to the migration framework (e.g. the introduction of adapters) haven't been propagated to this repo yet.

**Comment by eecavanna** [comment]
Side note: I'm surprised the new slot's name is singular instead of plural. I consider that topic to be outside the scope of this PR, though.

File: nmdc_schema/migrators/migrator_from_X_to_PR21.py (hunk 1 above)
Comment (brynnz22) [reply nmdc_schema/migrators/migrator_from_X_to_PR21.py:19]: Oops, this should not be here. I will fix the description. This is for an old migrator

**Comment by eecavanna** [comment]
This branch looks good to me, as of the latest commit.

**Review by turbomam — APPROVED** [review]

**Pull request merged**